### PR TITLE
Add support for the weight key added in nerve

### PR DESCRIPTION
**Pull request by bobtfish**

### CODE REVIEW

**Comment by bobtfish** [comment]
@jnb can you review this whilst @jolynch is out?

File: lib/synapse/service_watcher/zookeeper.rb [hunk 1]
```diff
@@ -70,7 +70,11 @@ def discover
           numeric_id = NUMBERS_RE =~ numeric_id ? numeric_id.to_i : nil
 
           log.debug "synapse: discovered backend #{name} at #{host}:#{server_port} for service #{@name}"
-          new_backends << { 'name' => name, 'host' => host, 'port' => server_port, 'id' => numeric_id}
+          new_backend = { 'name' => name, 'host' => host, 'port' => server_port, 'id' => numeric_id}
+          if !weight.nil?
```
Comment (jolynch) [inline lib/synapse/service_watcher/zookeeper.rb]: Should we keep the handling with name consistent (i.e. name will just be nil if it doesn't have a name, same for weight)?

**Comment by jolynch** [comment]
This just adds support for getting it into the backends themselves, but not for using that information in HAProxy right? Presumably that'll come later (it's a trickier patch)?

**Comment by bobtfish** [comment]
Yes, it just puts the info into the backends themselves (and therefore outputs them in the file output, which is what I need for extlbs).

The synapse managed haproxy will follow (as you note, it's not quite as easy, and we probably need a mechanism to tell synapse to ignore / override the nerve provided weights) - either way, it's likely to merit more discussion than this change, and so I split them up so that we can get this part in ASAP and unblock @vulpine's other work whilst we consider/discuss the patch to get the data into the haproxy config.

**Comment by jolynch** [comment]
Seems reasonable, merged this and let's chat about the next steps later. We have to remember to allow stats socket updates when we do the weight changes.

**Pull request merged**